### PR TITLE
First cloud implementation including transmission spectra

### PR DESCRIPTION
**Pull request by Kiefersv**
Hey all,

I finally come around and added the first cloud support routine to our nice little package! I kept it as general as possible so it should be compatible with other GCMs as well. In particluar I:
- Defined cloud properties
- Added cloud read in for clex/MITgcm (well clex is still in development but we getting there)
- Added transmission spectrum calculation for cloudy atmospheres using the prt interface
- Of course everything comes already neatly packed with pylint and ci checks.

Looking forward to your reviews!

### CODE REVIEW

File: gcm_toolkit/utils/clouds.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    """
+    This function is used to update petitRADTRANS to include clouds.
+    """
+    if clouds is not None:
+        lsk = self.line_struc_kappas
+        self.line_struc_kappas = np.zeros((lsk.shape[0], lsk.shape[1],
+                                           lsk.shape[2]+1, lsk.shape[3]))
+        self.line_struc_kappas[:, :, :-1, :] = lsk
+        for j in range(len(self.line_struc_kappas[:, 0, 0, 0])):
+            self.line_struc_kappas[j, :, -1, :] = clouds[0]
```
Comment (AaronDavidSchneider) [inline gcm_toolkit/utils/clouds.py:331]: such a clever function! I assume you tested it?

You could get rid of the loop using
```suggestion
self.line_struc_kappas[:, :, -1, :] = np.ones_like(self.line_struc_kappas[:, :, -1, :])*clouds[0][np.newaxis,:,:]
```
But yours is oc fine as well!

File: gcm_toolkit/exorad/read_in.py [hunk 2]
```diff
@@ -119,6 +187,15 @@ def m_read_from_mitgcm(
...
 
+    # change the names of the cloud variables
+    if 'ClAb' in dsi.keys():
+        dsi = dsi.rename({'Cl01': 'ClVf_TiO2[s]', 'Cl02': 'ClVf_Mg2SiO4[s]', 'Cl03': 'ClVf_SiO[s]',
+                          'Cl04': 'ClVf_SiO2[s]', 'Cl05': 'ClVf_Fe[s]', 'Cl06': 'ClVf_Al2O3[s]',
+                          'Cl07': 'ClVf_CaTiO3[s]', 'Cl08': 'ClVf_FeO[s]', 'Cl09': 'ClVf_FeS[s]',
+                          'Cl10': 'ClVf_Fe2O3[s]', 'Cl11': 'ClVf_MgO[s]', 'Cl12': 'ClVf_MgSiO3[s]',
+                          'Cl13': 'ClVf_CaSiO3[s]', 'Cl14': 'ClVf_Fe2SiO4[s]', 'Cl15': 'ClVf_C[s]',
+                          'Cl16': 'ClVf_KCl[s]'})
+
```
Comment (AaronDavidSchneider) [inline gcm_toolkit/exorad/read_in.py:198]: This makes no sense to me, what do you aim to achieve with it? Is this easier than just setting these keys above?

**Review by AaronDavidSchneider — COMMENTED** [review]
Hi, looks overall good to me. Is it allowed to have these data files public?
Also, could we do the same for the phasecurve?

File: gcm_toolkit/exorad/read_in.py (hunk 2 above)
Comment (Kiefersv) [reply gcm_toolkit/exorad/read_in.py:198]: Within the MITgcm, the cloud species are hard coded with numbers and have to stay that way.
Within gcmt, the idea is to have a more flexible approach to allow for multiple different species.

I did not find a way to read in, for example, Cl01 as ClVf_TiO2[s], but if you know how, please let me know!

File: gcm_toolkit/utils/clouds.py (hunk 1 above)
Comment (Kiefersv) [reply gcm_toolkit/utils/clouds.py:331]: If tested it on HATS-6b and the results look reasonable. I also run tests on submodules which should be correct. I did not yet benchmarked it against other cloud codes which might be difficult.

File: gcm_toolkit/exorad/read_in.py (hunk 2 above)
Comment (AaronDavidSchneider) [reply gcm_toolkit/exorad/read_in.py:198]: ahhhhh..... now i get it. Ok thanks!

**Comment by Kiefersv** [comment]
We double checked all data and everything is publicly available. Therefore, there is no problem in using it here as well.

If there are no further comments or concerns you can merge this branch.

**Pull request merged**